flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/JoinToMultiJoinRule.java

```diff
@@ -345,41 +327,25 @@ private List<RelNode> combineInputs(
      * The join type is also set.
      *
      * @param joinRel join rel
-     * @param combinedInputs the combined inputs to the join
      * @param left left child of the joinrel
-     * @param right right child of the joinrel
      * @param joinSpecs the list where the join types and conditions will be copied
      */
     private void combineJoinInfo(
-            Join joinRel,
-            List<RelNode> combinedInputs,
-            RelNode left,
-            RelNode right,
-            List<Pair<JoinRelType, RexNode>> joinSpecs,
-            List<Boolean> inputNullGenFieldList) {
+            Join joinRel, RelNode left, List<Pair<JoinRelType, RexNode>> joinSpecs) {
         JoinRelType joinType = joinRel.getJoinType();
-        JoinInfo joinInfo = joinRel.analyzeCondition();
-        ImmutableIntList leftKeys = joinInfo.leftKeys;
         final RexBuilder rexBuilder = joinRel.getCluster().getRexBuilder();
         boolean leftCombined = canCombine(left, joinRel);
         switch (joinType) {
             case LEFT:
-                if (leftCombined) {
-                    copyJoinInfo((MultiJoin) left, joinSpecs);
-                } else {
-                    joinSpecs.add(Pair.of(JoinRelType.INNER, rexBuilder.makeLiteral(true)));
-                }
-                joinSpecs.add(Pair.of(joinType, joinRel.getCondition()));
-                break;
             case INNER:
                 if (leftCombined) {
                     copyJoinInfo((MultiJoin) left, joinSpecs);
                 } else {
                     joinSpecs.add(Pair.of(JoinRelType.INNER, rexBuilder.makeLiteral(true)));
                 }
```

Comment:
may be I missed something however they are identical

Reply:
They were